coral/src/app/features/topics/request/TopicRequest.test.tsx

```diff
@@ -271,6 +272,59 @@ describe("<TopicRequest />", () => {
       });
     });
 
+    describe("informs user about valid input with a prefix is needed", () => {
+      beforeAll(() => {
+        mockGetEnvironmentsForTeam({
+          mswInstance: server,
+          response: {
+            data: [
+              createMockEnvironmentDTO({
+                name: "DEV",
+                id: "4",
+                params: {
+                  applyRegex: false,
+                  topicPrefix: ["prefix_"],
+                },
+              }),
+            ],
+          },
+        });
+        mockgetTopicAdvancedConfigOptions({
+          mswInstance: server,
+          response: {
+            data: defaultgetTopicAdvancedConfigOptionsResponse,
+          },
+        });
```

Comment:
I've seen now what looks to be 2 different approaches on mocking, this what looks to use some sort of mocked server and the likes of this:
```
jest.mock("src/domain/topic/topic-api.ts");
const mockGetTopicRequests = getTopicRequests as jest.MockedFunction<
  typeof getTopicRequests
>;
mockGetTopicRequests.mockResolvedValue(mockGetTopicRequestsResponse);
```

Is this intentional? If so, what's the benefit using one over the other. Thanks!

Reply:
Ah, yes! We tried msw mocks in the beginning (which was also very helpful for developing the UI while the API was not ready) and this test is one of the older ones that still has that mock. But we discovered that the mocks made the tests slower and, where we had to mock more different apis, more brittle (we had 1 or 2 that broke in the pipeline more often due to that). 

It also got to a point that it felt we're more testing the correctness of our mock server setup then actual code. For example when testing that clicking "page 2" on a paginated page -> returning a response for page 2 from the msw mock -> testing that user actually get's the right response. But that would have stayed true even if the real api endpoint would change. So checking that "clicking on page 2" sends a certain request to the api (aka api mock) is enough to ensure usage (bc our types make sure we can't build wrong requests). 

That's when we changed to using jest mocks!